### PR TITLE
Value to units for buildings constructable throughout nation

### PR DESCRIPTION
**Pull request by alberts2**
Newer version of #93

### CODE REVIEW

**Comment by Toffer90** [comment]
> I resolved the conflict in A_New_Dawn_GlobalDefines.xml and updated this branch.
> 
> Are there any other known issues with these changes?

There was a performance concern, not sure if that was resolved adequately or not.
Someone should profile it and find out.

**Comment by raxo2222** [comment]
It seems like it fell out.

**Comment by Toffer90** [comment]
> It seems like it fell out.
> 
> Its so old, that there are two conflicted files now

Wasn't very important anyhow, the conflicts should be easy to resolve, as this branch did very little and should be easy to understand if one compare it to the master version it branch out from where there's no conflicts.

File: Sources/CvCityAI.cpp [hunk 1]
```diff
@@ -4858,198 +4848,161 @@ UnitTypes CvCityAI::AI_bestUnitAI(UnitAITypes eUnitAI, int& iBestValue, bool bAs
...
 	}
 
-	CvCivilizationInfo& kCivilizationInfo = GC.getCivilizationInfo(getCivilizationType());
-	for (iI = 0; iI < GC.getNumUnitClassInfos(); iI++)
+	for (iI = 0; iI < GC.getNumUnitInfos(); iI++)
 	{
-		eLoopUnit = ((UnitTypes)(kCivilizationInfo.getCivilizationUnits(iI)));
+		eLoopUnit = (UnitTypes) iI;
 
 		if (eLoopUnit != NO_UNIT)
```
Comment (billw2012) [inline Sources/CvCityAI.cpp]: This is impossible. NO_UNIT is -1.

**Comment by billw2012** [comment]
To clarify, this change is intended to adjust the construction value of units based on what buildings they might be able to construct, which currently is not considered?

File: Sources/CvCityAI.cpp (hunk 1 above)
Comment (Toffer90) [reply Sources/CvCityAI.cpp]: True, my change made that if check redundant, I tried to change as little as possible code for it to be easier to understand the real changes.

**Pull request closed without merging**